roles/update/templates/l3_agent_stop_ping.sh.j2

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+#
+# Script which stops the ICMP connectivity check on validates that
+# there is no packet loss.
+set -eu -o pipefail
+
+# In seconds.
+LOSS_THRESHOLD=${1:-{{ loss_threshold|default(60)|int }}}
+LOSS_THRESHOLD_PERCENT=${2:-{{ loss_threshold_percent|default(1)|int }}}
+# useful for testing
+BASE_DIR="${3:-{{ cifmw_update_artifacts_basedir }}}"
+
+# Return total cut time in seconds
+function get_cut_time()
+{
+    local start=$(head -2 $PING_RESULT_LOG | tail -1 | grep -o '\[[^.]*' | tr -d '[[]]')
+    local end=$(tail -n 5 $PING_RESULT_LOG | head -1 | grep -o '\[[^.]*' | tr -d '[[]]')
+    awk -vstart=$start -vend=$end -vpr=${PING_RESULT} \
+        'BEGIN{res=(end - start)*pr/100;print res}'
+}
+
+# kill the ping process
+kill -s INT $(/usr/sbin/pidof ping)
+
+# print the ping results
+PING_RESULT_LOG=$(find "${BASE_DIR}" -maxdepth 1 -iname 'ping_results*' | sort | tail -1)
+tail -2 $PING_RESULT_LOG
+
+# if only one packet is lost, test succeeded, avoid false positive.
+TRANSMITTED=$(grep 'packet loss' $PING_RESULT_LOG | sed 's/^\([0-9]*\) packets transmitted,.*/\1/')
+RECEIVED=$(grep 'packet loss' $PING_RESULT_LOG | sed 's/.*packets transmitted, \([0-9]*\) received,.*/\1/')
+if [ $(( ${TRANSMITTED} - ${RECEIVED} )) -eq 1 ]; then
+    echo "Only one ping was not replied during the update/upgrade - test succeeded"
+    exit 0
+fi
```

Comment:
what about if there is no loss at all i.e. -eq 0 , it should still exit 0 ?

Reply:
This is handled in the following case.  I've improved the documentation around this.  Let me know if that's better.